Sources/Controllers/MyPlaces/TracksSearchFilter.swift

```diff
@@ -0,0 +1,321 @@
+                filter.doInitWithValue(value: selectedFilter)
+            }
+        }
+    }
+    
+    func onFilterChanged() {
+        for listener in filterChangedListeners {
+            listener.onFilterChanged()
+        }
+    }
+    
+    func resetFilteredItems() {
+        filteredTrackItems = []
+    }
+    
+    func getFilteredTrackItems() -> [TrackItem] {
+        filteredTrackItems
+    }
+    
+    func setFilteredTrackItems(_ trackItems: [TrackItem]) {
+        filteredTrackItems = trackItems
+    }
+    
+    func setAllItems(_ trackItems: [TrackItem]) {
+        self.trackItems = trackItems
+    }
+    
+    func getAllItems() -> [TrackItem] {
+        trackItems
+    }
+    
+    func setCurrentFolder(_ currentFolder: TrackFolder) {
+        self.currentFolder = currentFolder
+    }
+    
+    func getFilterSpecificSearchResults() -> [TrackFilterType: [TrackItem]] {
+        filterSpecificSearchResults
+    }
+}
+
+extension TracksSearchFilter {
+    static func getDisplayMinValue(filter: RangeTrackFilter<AnyObject>) -> Int {
+        let formattedValue = getFormattedValue(measureUnitType: filter.trackFilterType.measureUnitType, value: String(filter.ceilMinValue()))
+        return Int(formattedValue.valueSrc)
+    }
+    
+    static func getDisplayMaxValue(filter: RangeTrackFilter<AnyObject>) -> Int {
+        let formattedValue = getFormattedValue(measureUnitType: filter.trackFilterType.measureUnitType, value: filter.ceilMaxValue())
+        return Int(formattedValue.valueSrc)
+    }
+    
+    static func getDisplayValueFrom(filter: RangeTrackFilter<AnyObject>) -> Int {
+        let formattedValue = getFormattedValue(measureUnitType: filter.trackFilterType.measureUnitType, value: String(describing: filter.valueFrom))
+        return Int(formattedValue.valueSrc)
+    }
+    
+    static func getDisplayValueTo(filter: RangeTrackFilter<AnyObject>) -> Int {
+        let formattedValue = getFormattedValue(measureUnitType: filter.trackFilterType.measureUnitType, value: filter.ceilValueTo())
+        return Int(formattedValue.valueSrc)
+    }
+    
+    static func getFormattedValue(measureUnitType: MeasureUnitType, value: String) -> FormattedValue {
+        let metricsConstants = OAAppSettings.sharedManager().metricSystem.get()
+        let formattedString: String?
+        switch measureUnitType {
+        case .speed:
+            formattedString = OAOsmAndFormatter.getFormattedSpeed(Float(value) ?? 0.0)
+        case .altitude:
+            formattedString = OAOsmAndFormatter.getFormattedAlt(Double(value) ?? 0.0, mc: metricsConstants)
+        case .distance:
+            formattedString = OAOsmAndFormatter.getFormattedDistance(Float(value) ?? 0.0)
+        case .timeDuration:
+            let durationValue = Float(value) ?? 0.0
+            return FormattedValue(valueSrc: durationValue / 1000 / 60, value: String(format: "%.2f", durationValue / 1000 / 60), unit: "")
```

Comment:
Create a FormattedValue file and move FormattedValue class there

Reply:
The FormattedValue class has already been created